### PR TITLE
feat: add `blas/base/zswap`

### LINKED ISSUE #2047: [RFC]: add `blas/base/zswap`
### Description

This RFC proposes adding the package `@stdlib/blas/base/zswap` for interchanging two complex double-precision floating-point vectors.
```c
void c_zswap( const int N, void *X, const int strideX, void *Y, const int strideY )
```

### Related Issues

Related issues: #2039

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by performant23**
If applied, this commit will add the package `blas/base/zswap`

## Description

> What is the purpose of this pull request?
Add the package `blas/base/zswap`

This pull request:
- includes the support for `C` and `Fortran` and obtains Fortran reference from [this page](https://netlib.org/lapack/explore-html-3.6.1/d2/df9/group__complex16__blas__level1_ga13a187010a0cae1fef2820072404e857.html#ga13a187010a0cae1fef2820072404e857)

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #2047 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by performant23** [comment]
Refactoring the code to use `real` instead of `realf` as reviewed in `blas/base/zcopy`, will reopen once done.

**Comment by kgryte** [comment]
@performant23 Looks like you are missing the `Makefile` for the `/src` directory.

File: lib/node_modules/@stdlib/blas/base/zswap/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/README.md]:
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameters support indexing semantics based on starting indices. For example, to interchange every other value in `x` starting from the second value into the last `N` elements in `y` where `x[i] = y[n]`, `x[i+2] = y[n-1]`, and so on,
```

File: lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f]:
```suggestion
        complex(kind=kind(0.0d0)) :: zx(:), zy(:)
```

File: lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f]:
```suggestion
      x( i ) = cmplx( (r1*10000.0d0)-5000.0d0, (r2*10000.0d0)-5000.0d0, kind=kind(0.0d0) )
```

File: lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f:217]:
```suggestion
end program bench

```

File: lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/benchmark/fortran/benchmark.length.f]:
```suggestion
    complex(kind=kind(0.0d0)), allocatable :: x(:), y(:)
```

File: lib/node_modules/@stdlib/blas/base/zswap/package.json
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/package.json]:
```suggestion
    "double",
```

File: lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c]: Prefer matching how you implemented JS with only one `tmp` variable.

File: lib/node_modules/@stdlib/blas/base/zswap/src/zswap.f
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/src/zswap.f]:
```suggestion
! @param {Array<complex<double>>} zx - first input array
! @param {integer} strideX - `zx` stride length
! @param {Array<complex<double>>} zy - second input array
```

File: lib/node_modules/@stdlib/blas/base/zswap/src/zswap.f
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/src/zswap.f]:
```suggestion
  complex(kind=kind(0.0d0)), dimension(:) :: zx, zy
```

File: lib/node_modules/@stdlib/blas/base/zswap/src/zswap.f
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/src/zswap.f]:
```suggestion
  complex(kind=kind(0.0d0)) :: tmp
```

File: lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c]:
```suggestion
	double tmp;
```

File: lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c]:
```suggestion
			tmp = x[ i ];
			x[ i ] = y[ i ];
			y[ i ] = tmp;

			tmp = x[ i+1 ];
			x[ i+1 ] = y[ i+1 ];
			y[ i+1 ] = tmp;
```

File: lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zswap/src/zswap.c]:
```suggestion
		tmp = x[ ix ];
		x[ ix ] = y[ iy ];
		y[ iy ] = tmp;

		tmp = x[ ix+1 ];
		x[ ix+1 ] = y[ iy+1 ];
		y[ iy+1 ] = tmp;
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @performant23!

**Comment by kgryte** [comment]
Ah, wait. You need to add the missing `Makefile` before we can merge.

**Comment by performant23** [comment]
> Ah, wait. You need to add the missing `Makefile` before we can merge.

Added, thank you so much for your review!

Edit: Ah, some tests are failing, working on the fix

Hmm, that's strange. everything seems to work locally 
image

**Comment by kgryte** [comment]
@performant23 As you are on Windows, you were testing against the C implementation which did not have any issues. However, the Fortran implementation did have bugs, and the Fortran implementation was what was being tested against in CI. The issues should now be addressed.

**Review by kgryte — APPROVED** [review]
This LGTM. Thanks, @performant23! I'll merge once CI passes.

**Pull request merged**
